src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -565,11 +609,10 @@ private sealed class TokenOrderComparer : IComparer<SyntaxToken>
         {
             public static readonly TokenOrderComparer Instance = new TokenOrderComparer();
 
-            private TokenOrderComparer() { }
-
             public int Compare(SyntaxToken x, SyntaxToken y)
             {
-                return x.FullSpan.CompareTo(y.FullSpan);
+                int dpos = x.FullSpan.Start.CompareTo(y.FullSpan.Start);
+                return dpos != 0 ? dpos : x.FullSpan.End.CompareTo(y.FullSpan.End);
```

Comment:
interesting.  was this necessary?  if so, please place a comment explaining the case.

Reply:
Reverted, as a test showed, that is has little to no effect. Introduced for the same reason `Position` was introduced: e.g. for some reason profiler was showing `FullSpan` computation as hot.